react/components/RichTextEditor/index.jsx

```diff
@@ -0,0 +1,208 @@
+                    title="Ordered List"
+                    placement="top"
+                    enterDelay={1000}
+                    arrow
+                >
+                    <IconButton
+                        key={"orderedList"}
+                        onClick={() =>
+                            editor.chain().focus().toggleOrderedList().run()
+                        }
+                        className={
+                            editor.isActive("orderedList") ? "is-active" : ""
+                        }
+                    >
+                        <FormatListNumberedIcon />
+                    </IconButton>
+                </Tooltip>
+                <Tooltip
+                    data-testid="bulletedlist-tooltip"
+                    title="Bulleted List"
+                    placement="top"
+                    enterDelay={1000}
+                    arrow
+                >
+                    <IconButton
+                        key={"bulletList"}
+                        onClick={() =>
+                            editor.chain().focus().toggleBulletList().run()
+                        }
+                        className={
+                            editor.isActive("bulletList") ? "is-active" : ""
+                        }
+                        style={{ borderRight: "solid 1px #9c9c9c" }}
+                    >
+                        <FormatListBulletedIcon />
+                    </IconButton>
+                </Tooltip>
+                <Tooltip
+                    data-testid="table-tooltip"
+                    title="Table"
+                    placement="top"
+                    enterDelay={1000}
+                    arrow
+                >
+                    <IconButton
+                        key={"insertTable"}
+                        onClick={() =>
+                            editor.commands.insertTable({
+                                rows: 3,
+                                cols: 3,
+                                withHeaderRow: true,
+                            })
+                        }
+                        className={
+                            editor.isActive("insertTable") ? "is-active" : ""
+                        }
+                    >
+                        <TableChartIcon />
+                    </IconButton>
+                </Tooltip>
+            </div>
+        </div>
+    );
+};
+
+const RichTextEditor = ({ onChange, content }) => {
+    const editor = useEditor(
+        {
+            extensions: [
+                StarterKit,
+                Gapcursor,
+                Table.configure({
+                    resizable: true,
+                }),
+                TableRow,
+                TableHeader,
+                TableCell,
+                Underline,
+            ],
+            shouldRerenderOnTransaction: false,
+            content,
+            onUpdate: ({ editor }) => {
+                const newValue = editor.getHTML();
```

Comment:
can we get this in markdown instead of HTML. markdown makes sure everything is escaped properly. here's the markdown extension- https://github.com/aguingand/tiptap-markdown